gherkin/java/src/main/java/io/cucumber/gherkin/GherkinDialect.java

```diff
@@ -1,95 +1,109 @@
-
 import java.util.ArrayList;
 import java.util.List;
 
-import static java.util.Objects.requireNonNull;
+import static java.util.Collections.unmodifiableList;
 
 public final class GherkinDialect {
-    private final JsonObject keywords;
     private final String language;
-
-    public GherkinDialect(String language, JsonObject keywords) {
-        this.language = requireNonNull(language);
-        this.keywords = requireNonNull(keywords);
+    private final String name;
+    private final String nativeName;
+    private final List<String> featureKeywords;
+    private final List<String> ruleKeywords;
+    private final List<String> scenarioKeywords;
+    private final List<String> scenarioOutlineKeywords;
+    private final List<String> backgroundKeywords;
+    private final List<String> examplesKeywords;
+    private final List<String> givenKeywords;
+    private final List<String> whenKeywords;
+    private final List<String> thenKeywords;
+    private final List<String> andKeywords;
+    private final List<String> butKeywords;
+    private final List<String> stepKeywords;
+
+    GherkinDialect(String language, String name, String nativeName, List<String> featureKeywords, List<String> ruleKeywords, List<String> scenarioKeywords, List<String> scenarioOutlineKeywords, List<String> backgroundKeywords, List<String> examplesKeywords, List<String> givenKeywords, List<String> whenKeywords, List<String> thenKeywords, List<String> andKeywords, List<String> butKeywords) {
+        this.language = language;
+        this.name = name;
+        this.nativeName = nativeName;
+        this.featureKeywords = featureKeywords;
+        this.ruleKeywords = ruleKeywords;
+        this.scenarioKeywords = scenarioKeywords;
+        this.scenarioOutlineKeywords = scenarioOutlineKeywords;
+        this.backgroundKeywords = backgroundKeywords;
+        this.examplesKeywords = examplesKeywords;
+        this.givenKeywords = givenKeywords;
+        this.whenKeywords = whenKeywords;
+        this.thenKeywords = thenKeywords;
+        this.andKeywords = andKeywords;
+        this.butKeywords = butKeywords;
+
+        List<String> stepKeywords = new ArrayList<>();
+        stepKeywords.addAll(givenKeywords);
+        stepKeywords.addAll(whenKeywords);
+        stepKeywords.addAll(thenKeywords);
+        stepKeywords.addAll(andKeywords);
+        stepKeywords.addAll(butKeywords);
+        this.stepKeywords = unmodifiableList(stepKeywords);
     }
 
     public List<String> getFeatureKeywords() {
-        return toStringList(this.keywords.get("feature").asArray());
-    }
-
-    private static List<String> toStringList(JsonArray array) {
-        List<String> result = new ArrayList<>();
-        for (JsonValue jsonValue : array) {
-            result.add(jsonValue.asString());
-        }
-        return result;
+        return featureKeywords;
     }
 
     public String getName() {
-        return keywords.getString("name", null);
```

Comment:
Was this field nullable?

Reply:
No, all the entries in the JSON file have a `name` property.